### PR TITLE
Pass opts argument to api client in ruby-client

### PR DESCRIPTION
**Pull request by ota42y**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

### Description of the PR

We want to change the option when debugging, changing timeout to specific api, etc...
So we want to merge and pass options to api client.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/ruby-client/api.mustache [hunk 1]
```diff
@@ -173,20 +173,24 @@ module {{moduleName}}
...
-      data, status_code, headers = @api_client.call_api(:{{httpMethod}}, local_var_path,
+      post_body = opts[:body] {{#bodyParam}}|| @api_client.object_to_http_body({{#required}}{{{paramName}}}{{/required}}{{^required}}opts[:'{{{paramName}}}']{{/required}}) {{/bodyParam}}
+
+      # return_type
+      return_type = opts[:return_type] {{#returnType}}|| '{{{returnType}}}' {{/returnType}}
+
+      # auth_names
+      auth_names = opts[:auth_names] || [{{#authMethods}}'{{name}}'{{#hasMore}}, {{/hasMore}}{{/authMethods}}]
+
+      new_optios = opts.merge(
```
Comment (autopp) [inline modules/openapi-generator/src/main/resources/ruby-client/api.mustache]: Is this a typo of `options`?

Comment (ota42y) [reply modules/openapi-generator/src/main/resources/ruby-client/api.mustache]: I fixed! 🙇

Comment (autopp) [reply modules/openapi-generator/src/main/resources/ruby-client/api.mustache]: 👍

**Comment by wing328** [comment]
@ota42y thanks again for the PR. When you've time, I wonder if you can PM me via [Slack](https://join.slack.com/t/openapi-generator/shared_invite/enQtNzAyNDMyOTU0OTE1LTY5ZDBiNDI5NzI5ZjQ1Y2E5OWVjMjZkYzY1ZGM2MWQ4YWFjMzcyNDY5MGI4NjQxNDBiMTlmZTc5NjY2ZTQ5MGM) or [Twitter](https://twitter.com/wing328) as I've some questions on this PR.

**Pull request merged**